src/libsyntax/ext/expand.rs

```diff
@@ -514,8 +514,12 @@ impl<'a> Parser<'a> {
             ExpansionKind::Stmts => {
                 let mut stmts = SmallVector::zero();
                 while self.token != token::Eof {
```

Comment:
I believe we could just change this line to
```rust
while self.token != token::Eof && self.token != token::CloseDelim(token::Brace) {
```
instead of checking `token_consumed` for progress.